From 58774f57d5e46df1da58a02c8bbf10784d32e4f8 Mon Sep 17 00:00:00 2001
From: ningyu <ningyu@eswincomputing.com>
Date: Fri, 21 Feb 2025 17:35:41 +0800
Subject: [PATCH 318/533] feat:support osm dts

Changelogs:
1. support osm dts
2. osm eth0
3. osm add codec
4. emmc disable write protect

Signed-off-by: ningyu <ningyu@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7700-osm.dts | 195 ++++++++--------------
 arch/riscv/configs/eic7700_defconfig      |  64 ++++++-
 2 files changed, 131 insertions(+), 128 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-osm.dts b/arch/riscv/boot/dts/eswin/eic7700-osm.dts
index ab2bf385a71e..5e9d67c18185 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-osm.dts
+++ b/arch/riscv/boot/dts/eswin/eic7700-osm.dts
@@ -33,6 +33,7 @@
 #include <dt-bindings/interrupt-controller/irq.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
+//#include "eic7700-osm-camera.dtsi"
 
 /* Clock frequency (in Hz) of the PCB crystal for rtcclk */
 
@@ -40,8 +41,7 @@ / {
 	#address-cells = <2>;
 	#size-cells = <2>;
 	model = "ESWIN EIC7700 OSM";
-	compatible = "sifive,hifive-unmatched-a00", "sifive,fu740-c000",
-		     "sifive,fu740", "eswin,eic7700";
+	compatible = "eswin,eic7x";
 
 	aliases {
 		serial0 = &d0_uart0;
@@ -93,9 +93,14 @@ npu0_reserved: sprammemory@59000000 {
 			reg = <0x0 0x59000000 0x0 0x400000>;
 		};
 
-		g2d_4GB_boundary_reserved_4k {
+		ramoops@fff00000 {
+			compatible = "ramoops";
+			reg = <0x0 0xfff00000 0x0 0x100000>;
+			console-size = <0x40000>;
+			record-size = <0x40000>;
+			ftrace-size = <0x40000>;
+			pmsg-size = <0x40000>;
 			no-map;
-			reg = <0x0 0xfffff000 0x0 0x1000>;
 		};
 
 		g2d_8GB_boundary_reserved_4k {
@@ -114,37 +119,19 @@ mmz_nid_0_part_0 {
 			no-map;
 		};
 	};
+	reg_vcc1v8: vcc1v8 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc1v8";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+	};
 };
 
 &d0_clock {
 	status = "okay";
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_gpio28_default>;
-	cpu-voltage-gpios = <&porta 28 GPIO_ACTIVE_HIGH>;
-	cpu-default-frequency = <CLK_FREQ_1600M>;
-};
-
-&d0_cpu_opp_table {
-	opp-1500000000 {
-		opp-hz = /bits/ 64 <CLK_FREQ_1500M>;
-		opp-microvolt = <900000>;
-		clock-latency-ns = <70000>;
-	};
-	opp-1600000000 {
-		opp-hz = /bits/ 64 <CLK_FREQ_1600M>;
-		opp-microvolt = <900000>;
-		clock-latency-ns = <70000>;
-	};
-	opp-1700000000 {
-		opp-hz = /bits/ 64 <CLK_FREQ_1700M>;
-		opp-microvolt = <900000>;
-		clock-latency-ns = <70000>;
-	};
-	opp-1800000000 {
-		opp-hz = /bits/ 64 <CLK_FREQ_1800M>;
-		opp-microvolt = <900000>;
-		clock-latency-ns = <70000>;
-	};
+	pinctrl-0 = <&pinctrl_gpio94_default>;
+	cpu-voltage-gpios = <&portc 30 GPIO_ACTIVE_HIGH>;
 };
 
 &d0_reset {
@@ -317,10 +304,6 @@ &dc {
 	dc_out: port {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		dc_out_dsi: endpoint@0 {
-			reg = <0>;
-			remote-endpoint = <&dsi_input>;
-		};
 		dc_out_dpi1: endpoint@1 {
 			reg = <1>;
 			remote-endpoint = <&vd_input>;
@@ -349,50 +332,6 @@ &dsi_output {
 	status = "okay";
 };
 
-&dsi_controller {
-	status = "okay";
-	ports {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		port@0 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <0>;
-			dsi_input: endpoint {
-				remote-endpoint = <&dc_out_dsi>;
-			};
-		};
-		port@1 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <1>;
-			mipi_dsi_out: endpoint {
-				remote-endpoint = <&panel_in>;
-			};
-		};
-	};
-
-	dsi_panel:dsi_panel@0 {
-		compatible = "eswin,generic-panel";
-		reg = <0>;
-		dsi,format = <0>; //RGB888:0, RGB666:1, RGB565:3
-		dsi,lanes = <4>;
-		port {
-			panel_in: endpoint {
-				remote-endpoint = <&mipi_dsi_out>;
-			};
-		};
-	};
-};
-
-&dsi_panel {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_gpio5_default &pinctrl_gpio111_default>;
-	backlight0-gpios = <&porta 5 GPIO_ACTIVE_HIGH>;
-	rst-gpios = <&portd 15 GPIO_ACTIVE_HIGH>;
-};
-
 &dw_hdmi {
 	status = "okay";
 	ports {
@@ -459,38 +398,22 @@ &d0_graphcard2 {
 	status = "disabled";
 };
 
-&isp_0 {
-	status = "okay";
-};
-
-&isp_1 {
+&vi_subsys {
 	status = "okay";
 };
 
-&dewarp {
-	status = "okay";
-};
-
-&mipi_dphy_rx {
-	status = "okay";
-};
-
-&csi_dma0 {
-	status = "okay";
+&isp_0 {
+	status = "disabled";
 };
 
-&csi_dma1 {
+&isp_1 {
 	status = "disabled";
 };
 
-&csi2_0 {
+&dewarp {
 	status = "okay";
 };
 
-&csi2_1 {
-	status = "disabled";
-};
-
 &sdhci_emmc {
 	/* emmc */
 	status = "okay";
@@ -515,6 +438,9 @@ &sdio0 {
 	disable-wp;
 	no-sdio;
 	no-mmc;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio13_default>;
+	cd-gpios = <&porta 13 GPIO_ACTIVE_LOW>;
 };
 
 &sdio1 {
@@ -528,6 +454,7 @@ &sdio1 {
 	non-removable;
 	no-sd;
 	no-mmc;
+	vqmmc-supply = <&reg_vcc1v8>;
 	aw3155:wifi_aw3155@0 {
 		compatible = "aml_w1_sdio";
 		reg = <0x0>;
@@ -540,6 +467,12 @@ aw3155:wifi_aw3155@0 {
 		rst-gpios = <&portb 2 GPIO_ACTIVE_HIGH>;
 
 	};
+	mdhd_wlan {
+		compatible = "android,bcmdhd_wlan";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_gpio34_default>;
+		gpio_wl_reg_on = <&portb 2 GPIO_ACTIVE_HIGH>;
+	};
 };
 
 &d0_gmac0 {
@@ -550,6 +483,7 @@ &d0_gmac0 {
 	eswin,rgmiisel = <&pinctrl 0x290 0x3>;
 	eswin,led-cfgs = <0x6100 0xa40 0x420>;
 	eswin,phyaddr = <1>;
+	phy-mode = "rgmii-txid";
 	dly-param-1000m = <0x20202020 0x64207820 0x20202020>;
 	dly-param-100m = <0x20202020 0x6E206E20 0x20202020>;
 	dly-param-10m = <0 0 0>;
@@ -563,6 +497,7 @@ &d0_gmac1 {
 	rst-gpios = <&porta 16 GPIO_ACTIVE_LOW>;
 	eswin,rgmiisel = <&pinctrl 0x294 0x3>;
 	eswin,led-cfgs = <0x6100 0xa40 0x420>;
+	phy-mode = "rgmii-txid";
 	dly-param-1000m = <0x25252525 0x80268025 0x26262626>;
 	dly-param-100m = <0x48484848 0x80588048 0x58585858>;
 	dly-param-10m = <0 0 0>;
@@ -623,6 +558,8 @@ &d0_uart2 {
 
 &d0_uart3 {
 	/* pin header */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3_default>;
 	status = "okay";
 };
 
@@ -632,8 +569,18 @@ &d0_uart4 {
 };
 
 &ssi0 {
-	/* unused */
-	status = "disabled";
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1_default>;
+	num-cs = <2>;
+	spi-flash@1 {
+		compatible = "jedec,spi-nor";
+		reg = <1>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <4800000>;
+		rx-sample-delay-ns = <10>;
+	};
 };
 
 &ssi1 {
@@ -645,8 +592,6 @@ &bootspi {
 	/* spi flash */
 	status = "okay";
 	num-cs = <1>;
-	cs-gpios = <&portd 0 GPIO_ACTIVE_LOW>;
-	wp-gpios = <&portd 4 GPIO_ACTIVE_LOW>;
 	spi-flash@0 {
 		compatible = "winbond,w25q128jw",
 					"jedec,spi-nor";
@@ -692,11 +637,19 @@ &d0_mbox7 {
 
 &fan_control {
 	status = "okay";
-	eswin,pwm_inverted;
+	pwms = <&pwm0 0 100000 PWM_POLARITY_INVERTED>;
 };
 
 &d0_i2c0 {
-	status = "disabled";
+	status = "okay";
+	eeprom@51 {
+		compatible = "atmel,24c64";
+		reg = <0x51>;
+	};
+	temp-sense@4e {
+		compatible = "national,lm75";
+		reg = <0x4e>;
+	};
 };
 
 &d0_i2c1 {
@@ -723,6 +676,10 @@ d0_codec3106_endpoint: endpoint {
 			};
 		};
 	};
+	rtc@32 {
+		compatible = "epson,rx8010";
+		reg = <0x32>;
+	};
 };
 
 &d0_i2c3 {
@@ -772,7 +729,7 @@ es5340@f {
 		compatible = "einno,es5340";
 		reg = <0xf>;
 		eswin,regulator_default-microvolt=<1000000>;
-		eswin,regulator_label = "supply vdd1", "npu vdd1", "npu current1", "npu temperature1";
+		eswin,regulator_label = "Supply VDD", "NPU Voltage", "NPU CURRENT", "NPU Temperature";
 		label = "npu_vdd";
 		regulators{
 			npu_vcc1:npu_svcc{
@@ -797,7 +754,7 @@ &pwm0 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&pinctrl_pwm0_default>;
 	pinctrl-1 = <&pinctrl_gpio68_default>;
-	enable-gpios = <&portc 4 GPIO_ACTIVE_HIGH>;
+	fan-pwm-gpios = <&portc 4 GPIO_ACTIVE_HIGH>;
 };
 
 &pvt0 {
@@ -828,26 +785,15 @@ &die0_rtc {
 	status = "okay";
 };
 
-&timer0 {
-	status = "okay";
-};
-
-&timer1 {
-	status = "okay";
-};
-
-&timer2 {
-	status = "okay";
-};
-
-&timer3 {
-	status = "okay";
-};
-
 &pinctrl {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio7_default &pinctrl_gpio8_default &pinctrl_gpio9_default &pinctrl_gpio10_default
+			&pinctrl_gpio71_default &pinctrl_gpio73_default &pinctrl_gpio74_default &pinctrl_gpio75_default
+			&pinctrl_gpio76_default &pinctrl_gpio77_default &pinctrl_gpio79_default &pinctrl_gpio80_default
+			&pinctrl_gpio83_default &pinctrl_gpio84_default &pinctrl_gpio85_default &pinctrl_gpio86_default
+			&pinctrl_gpio87_default &pinctrl_gpio89_default &pinctrl_gpio91_default>;
 };
-
 &gpio0 {
 	status = "okay";
 };
@@ -856,3 +802,4 @@ &dev_llc_d0{
 	npu-supply=<&npu_vcc1>;
 	status = "okay";
 };
+
diff --git a/arch/riscv/configs/eic7700_defconfig b/arch/riscv/configs/eic7700_defconfig
index f90fd076f227..964166f6fdee 100644
--- a/arch/riscv/configs/eic7700_defconfig
+++ b/arch/riscv/configs/eic7700_defconfig
@@ -29,6 +29,7 @@ CONFIG_BLK_DEV_INITRD=y
 CONFIG_EXPERT=y
 # CONFIG_SYSFS_SYSCALL is not set
 CONFIG_PERF_EVENTS=y
+CONFIG_KEXEC=y
 CONFIG_SOC_ESWIN=y
 CONFIG_SOC_VIRT=y
 CONFIG_SMP=y
@@ -442,6 +443,8 @@ CONFIG_SERIO_LIBPS2=y
 CONFIG_SERIAL_8250=y
 CONFIG_SERIAL_8250_CONSOLE=y
 CONFIG_SERIAL_8250_DW=y
+CONFIG_SERIAL_8250_NR_UARTS=5
+CONFIG_SERIAL_8250_RUNTIME_UARTS=5
 CONFIG_SERIAL_OF_PLATFORM=y
 CONFIG_SERIAL_EARLYCON_RISCV_SBI=y
 CONFIG_HVC_RISCV_SBI=y
@@ -471,6 +474,7 @@ CONFIG_SENSORS_ESWIN_PVT=y
 CONFIG_SENSORS_INA2XX=y
 CONFIG_SENSORS_TMP102=y
 CONFIG_SENSORS_PAC1934=y
+CONFIG_SENSORS_LM75=y
 CONFIG_THERMAL=y
 CONFIG_THERMAL_STATISTICS=y
 CONFIG_THERMAL_WRITABLE_TRIPS=y
@@ -482,11 +486,13 @@ CONFIG_DEVFREQ_THERMAL=y
 CONFIG_THERMAL_EMULATION=y
 CONFIG_WATCHDOG=y
 CONFIG_DW_WATCHDOG=y
+CONFIG_ESWIN_WATCHDOG=y
 CONFIG_REGULATOR=y
 CONFIG_REGULATOR_MPQ8785=y
 CONFIG_REGULATOR_TPS549D22=y
 CONFIG_REGULATOR_ES5340=y
 CONFIG_REGULATOR_PCA9450=y
+CONFIG_REGULATOR_RASPBERRYPI_TOUCHSCREEN_ATTINY=y
 CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_SUPPORT_FILTER=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
@@ -650,7 +656,11 @@ CONFIG_DRM_SIMPLE_BRIDGE=y
 CONFIG_DRM_TOSHIBA_TC358768=m
 CONFIG_DRM_SIMPLEDRM=m
 CONFIG_DRM_ESWIN=y
-CONFIG_ESWIN_MMU=y
+CONFIG_DRM_PANEL_SIMPLE=y
+CONFIG_DRM_TOSHIBA_TC358762=y
+CONFIG_DRM_DW_MIPI_DSI=y
+CONFIG_ESWIN_MIPI_DSI=y
+CONFIG_ESWIN_RASPBERRYPI_7INCH_PANEL=y
 CONFIG_ESWIN_DW_HDMI=y
 CONFIG_ESWIN_VIRTUAL_DISPLAY=y
 CONFIG_DW_HDMI_I2S_AUDIO=y
@@ -660,6 +670,16 @@ CONFIG_DRM_LEGACY=y
 CONFIG_FB=y
 CONFIG_FB_RADEON=m
 CONFIG_FB_SIMPLE=m
+CONFIG_VGA_CONSOLE=y
+CONFIG_DUMMY_CONSOLE=y
+CONFIG_DUMMY_CONSOLE_COLUMNS=80
+CONFIG_DUMMY_CONSOLE_ROWS=25
+CONFIG_FRAMEBUFFER_CONSOLE=y
+CONFIG_FRAMEBUFFER_CONSOLE_DETECT_PRIMARY=y
+CONFIG_LOGO=y
+CONFIG_LOGO_LINUX_MONO=y
+CONFIG_LOGO_LINUX_VGA16=y
+CONFIG_LOGO_LINUX_CLUT224=y
 CONFIG_SOUND=y
 CONFIG_SND=y
 CONFIG_SND_HDA_INTEL=m
@@ -667,6 +687,8 @@ CONFIG_SND_HDA_PATCH_LOADER=y
 CONFIG_SND_HDA_CODEC_REALTEK=m
 CONFIG_SND_HDA_CODEC_HDMI=m
 CONFIG_SND_HDA_PREALLOC_SIZE=2048
+CONFIG_SND_USB_AUDIO=y
+CONFIG_SND_USB_AUDIO_MIDI_V2=y
 CONFIG_SND_SOC=y
 CONFIG_SND_SOC_SOF_TOPLEVEL=y
 CONFIG_SND_SOC_SOF_OF=y
@@ -676,6 +698,7 @@ CONFIG_ESWIN_SND_ES8388_CODEC=y
 CONFIG_ESWIN_SND_TLV320AIC3106_CODEC=y
 CONFIG_SND_SIMPLE_CARD=y
 CONFIG_SND_AUDIO_GRAPH_CARD=y
+CONFIG_UHID=y
 CONFIG_USB_ULPI_BUS=y
 CONFIG_USB_CONN_GPIO=y
 CONFIG_USB=y
@@ -705,7 +728,7 @@ CONFIG_USB_SERIAL_PL2303=y
 CONFIG_USB_SERIAL_OTI6858=y
 CONFIG_USB_SERIAL_QUALCOMM=y
 CONFIG_USB_SERIAL_OPTION=y
-CONFIG_USB_TEST=y
+CONFIG_USB_TEST=m
 CONFIG_USB_GADGET=y
 CONFIG_USB_GADGET_DEBUG_FILES=y
 CONFIG_USB_GADGET_VBUS_DRAW=500
@@ -737,6 +760,7 @@ CONFIG_RTC_CLASS=y
 CONFIG_RTC_DRV_PCF85063=y
 CONFIG_RTC_DRV_PCF8563=y
 CONFIG_RTC_DRV_ESWIN=y
+CONFIG_RTC_DRV_RX8010=y
 CONFIG_DMADEVICES=y
 CONFIG_DW_AXI_DMAC=y
 CONFIG_DMATEST=m
@@ -766,7 +790,7 @@ CONFIG_ESWIN_DSP=m
 CONFIG_ESWIN_NPU=m
 CONFIG_ESWIN_BUS_ERROR=y
 CONFIG_PM=y
-CONFIG_PM_SLEEP=y
+# CONFIG_PM_SLEEP=y
 CONFIG_PM_DEVFREQ=y
 CONFIG_DEVFREQ_GOV_USERSPACE=y
 CONFIG_DEVFREQ_GOV_PERFORMANCE=y
@@ -781,6 +805,7 @@ CONFIG_ESWIN_MMZ_VB=y
 CONFIG_ESWIN_DEV_DMA_BUF=y
 CONFIG_ESWIN_IOMMU_RSV=y
 CONFIG_ESWIN_DMA_MEMCP=y
+CONFIG_ESWIN_MALLOC_DMABUF=y
 CONFIG_PWM=y
 CONFIG_PWM_ESWIN=y
 CONFIG_RESET_ESWIN=y
@@ -814,6 +839,11 @@ CONFIG_SQUASHFS_LZ4=y
 CONFIG_SQUASHFS_LZO=y
 CONFIG_SQUASHFS_XZ=y
 CONFIG_SQUASHFS_ZSTD=y
+CONFIG_PSTORE=y
+CONFIG_PSTORE_CONSOLE=y
+CONFIG_PSTORE_PMSG=y
+CONFIG_PSTORE_FTRACE=y
+CONFIG_PSTORE_RAM=y
 CONFIG_NFS_FS=y
 CONFIG_NFS_V3_ACL=y
 CONFIG_NFS_V4=y
@@ -832,16 +862,18 @@ CONFIG_CRYPTO_CBC=m
 # CONFIG_CRYPTO_XTS is not set
 CONFIG_CRYPTO_MD5=y
 CONFIG_CRYPTO_DEV_VIRTIO=y
+CONFIG_HW_RANDOM_ESWIN=y
 CONFIG_CRYPTO_DEV_ESWIN_EIC770x=y
 CONFIG_CRC_ITU_T=y
 CONFIG_CRC7=y
 CONFIG_DMA_CMA=y
 CONFIG_PRINTK_TIME=y
 CONFIG_CONSOLE_LOGLEVEL_DEFAULT=15
-CONFIG_CONSOLE_LOGLEVEL_QUIET=15
+CONFIG_CONSOLE_LOGLEVEL_QUIET=7
 CONFIG_MESSAGE_LOGLEVEL_DEFAULT=7
 CONFIG_DYNAMIC_DEBUG=y
 CONFIG_DEBUG_FS=y
+CONFIG_MAGIC_SYSRQ=y
 CONFIG_DEBUG_PAGEALLOC=y
 CONFIG_SCHED_STACK_END_CHECK=y
 CONFIG_DEBUG_VM=y
@@ -864,7 +896,31 @@ CONFIG_RCU_EQS_DEBUG=y
 # CONFIG_FTRACE is not set
 # CONFIG_RUNTIME_TESTING_MENU is not set
 CONFIG_MEMTEST=y
+CONFIG_VIDEO_ESWIN_VI_TOP=y
+CONFIG_VIDEO_ESWIN_DVP2AXI_DEPENDENCES=y
+CONFIG_VIDEO_ESWIN_DVP2AXI=y
+CONFIG_VIDEO_ESWIN_ISP=m
+CONFIG_DWC_MIPI_CSI2_HOST=y
+CONFIG_PHY_ESWIN_CSI2_DPHY=y
+CONFIG_OF_OVERLAY=y
+CONFIG_CONFIGFS_FS=y
+CONFIG_OF_CONFIGFS=y
+CONFIG_VIDEO_IMX219=m
+CONFIG_VIDEO_IMX258=m
+CONFIG_VIDEO_IMX290=m
+CONFIG_VIDEO_IMX327=m
+CONFIG_VIDEO_IMX415=m
+CONFIG_VIDEO_IMX586=m
+CONFIG_VIDEO_IMX708=m
+CONFIG_VIDEO_OV5647=m
+CONFIG_VIDEO_OV9282=m
+CONFIG_VIDEO_LT6911UXE=m
 CONFIG_I2C_MUX=y
 CONFIG_I2C_MUX_PCA954x=y
 CONFIG_BCMDHD=m
 CONFIG_BCMDHD_SDIO_IRQ=y
+CONFIG_REGULATOR=y
+CONFIG_REGULATOR_FIXED_VOLTAGE=y
+CONFIG_BCMDHD_FW_PATH="/lib/firmware/fw_bcm43456c5_ag.bin"
+CONFIG_BCMDHD_NVRAM_PATH="/lib/firmware/nvram_ap6256.txt"
+CONFIG_BCMDHD_CLM_PATH="/lib/firmware/clm_bcm43752a2_ag.blob"
-- 
2.52.0

